### PR TITLE
8315834: Open source several Swing JSpinner related tests 

### PR DESCRIPTION
**Pull request by TejeshR13**
Open source these Swing JSpinner related tests:

javax/swing/JSpinner/4522737/bug4522737.java
javax/swing/JSpinner/4656590/bug4656590.java
javax/swing/JSpinner/4680204/bug4680204.java
javax/swing/JSpinner/4862257/bug4862257.java
javax/swing/JSpinner/5104421/bug5104421.java

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315834](https://bugs.openjdk.org/browse/JDK-8315834): Open source several Swing JSpinner related tests (**Bug** - P4)

### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) ⚠️ Review applies to [094b7a32](https://git.openjdk.org/jdk/pull/15678/files/094b7a32ba2b4f607aef834b8b139ae59af66f9b)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15678/head:pull/15678` \
`$ git checkout pull/15678`

Update a local copy of the PR: \
`$ git checkout pull/15678` \
`$ git pull https://git.openjdk.org/jdk.git pull/15678/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15678`

View PR using the GUI difftool: \
`$ git pr show -t 15678`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15678.diff">https://git.openjdk.org/jdk/pull/15678.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15678#issuecomment-1715019965)

### CODE REVIEW

File: test/jdk/javax/swing/JSpinner/bug4522737.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            try {
+                ByteArrayOutputStream byteArrayOutputStream =
+                        new ByteArrayOutputStream();
+                ObjectOutputStream objectOutputStream =
+                        new ObjectOutputStream(byteArrayOutputStream);
+                objectOutputStream.writeObject(originalComponent);
+
+                objectOutputStream = new ObjectOutputStream(byteArrayOutputStream);
+                objectOutputStream.writeObject(originalComponent);
+            } catch (Throwable e) {
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JSpinner/bug4522737.java:51]: You can remove this try-catch block and let original exception propagate

File: test/jdk/javax/swing/JSpinner/bug4680204.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+                sp1 = new JSpinner(new SpinnerNumberModel(1, 1, 100, 1));
+                sp1.setToolTipText(TOOL_TIP_TEXT);
+                frame.getContentPane().add(sp1);
+
+                sp2 = new JSpinner();
+                sp2.setToolTipText(TOOL_TIP_TEXT);
+                frame.getContentPane().add(sp2);
+                sp2.setModel(new SpinnerNumberModel(1, 1, 100, 1));
+                frame.pack();
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JSpinner/bug4680204.java:65]: Probably setLocationRelativeTo call would be good

**Review by prsadhuk — APPROVED** [review]

File: test/jdk/javax/swing/JSpinner/bug4522737.java (hunk 1 above)
Comment (TejeshR13) [reply test/jdk/javax/swing/JSpinner/bug4522737.java:51]: The test can be headless right? Not necessary to be headful?

Comment (prsadhuk) [reply test/jdk/javax/swing/JSpinner/bug4522737.java:51]: yes

File: test/jdk/javax/swing/JSpinner/bug4656590.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            });
+            robot.waitForIdle();
+            robot.delay(1000);
+            if (failed) {
+                throw new RuntimeException("JSpinner.setFont() " +
+                        "doesn't set the font properly");
+            }
+            System.out.println("Test Passed!");
+        } finally {
+            if (frame != null) {
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JSpinner/bug4656590.java:110]: I think there is a discussion going on to keep this `frame null check` also inside SwingUtilities. 

Are you going to update here as well?

Comment (TejeshR13) [reply test/jdk/javax/swing/JSpinner/bug4656590.java:110]: Yes.

**Review by kumarabhi006 — APPROVED** [review]
LGTM.  Assuming you have run the CI tests after moving to open repo.

**Pull request closed without merging**